cmd/floating_ips_cmd.go

```diff
@@ -0,0 +1,201 @@
+func WireUpFloatingIP(c *cli.Context) (ds *network.FloatingIPService, f format.Formatter) {
+
+	f = format.GetFormatter()
+
+	config, err := utils.GetConcertoConfig()
+	if err != nil {
+		f.PrintFatal("Couldn't wire up config", err)
+	}
+	hcs, err := utils.NewHTTPConcertoService(config)
+	if err != nil {
+		f.PrintFatal("Couldn't wire up concerto service", err)
+	}
+	ds, err = network.NewFloatingIPService(hcs)
+	if err != nil {
+		f.PrintFatal("Couldn't wire up floating IP service", err)
+	}
+
+	return ds, f
+}
+
+// FloatingIPList subcommand function
+func FloatingIPList(c *cli.Context) error {
+	debugCmdFuncInfo(c)
+	floatingIPSvc, formatter := WireUpFloatingIP(c)
+
+	floatingIPs, err := floatingIPSvc.GetFloatingIPList(c.String("server-id"))
+	if err != nil {
+		formatter.PrintFatal("Couldn't receive floating IP data", err)
+	}
+
+	labelables := make([]types.Labelable, len(floatingIPs))
+	for i := 0; i < len(floatingIPs); i++ {
+		labelables[i] = types.Labelable(floatingIPs[i])
+	}
+	labelIDsByName, labelNamesByID := LabelLoadsMapping(c)
+	filteredLabelables := LabelFiltering(c, labelables, labelIDsByName)
+	LabelAssignNamesForIDs(c, filteredLabelables, labelNamesByID)
+
+	floatingIPs = make([]*types.FloatingIP, len(filteredLabelables))
+	for i, labelable := range filteredLabelables {
+		fw, ok := labelable.(*types.FloatingIP)
+		if !ok {
+			formatter.PrintFatal("Label filtering returned unexpected result",
+				fmt.Errorf("expected labelable to be a *types.FloatingIP, got a %T", labelable))
+		}
+		floatingIPs[i] = fw
+	}
+	if err = formatter.PrintList(floatingIPs); err != nil {
+		formatter.PrintFatal("Couldn't print/format result", err)
+	}
+	return nil
+}
+
+// FloatingIPShow subcommand function
+func FloatingIPShow(c *cli.Context) error {
+	debugCmdFuncInfo(c)
+	floatingIPSvc, formatter := WireUpFloatingIP(c)
+
+	checkRequiredFlags(c, []string{"id"}, formatter)
+	floatingIP, err := floatingIPSvc.GetFloatingIP(c.String("id"))
+	if err != nil {
+		formatter.PrintFatal("Couldn't receive floating IP data", err)
+	}
+	_, labelNamesByID := LabelLoadsMapping(c)
+	floatingIP.FillInLabelNames(labelNamesByID)
+	if err = formatter.PrintItem(*floatingIP); err != nil {
+		formatter.PrintFatal("Couldn't print/format result", err)
+	}
+	return nil
+}
+
+// FloatingIPCreate subcommand function
+func FloatingIPCreate(c *cli.Context) error {
+	debugCmdFuncInfo(c)
+	floatingIPSvc, formatter := WireUpFloatingIP(c)
+
+	checkRequiredFlags(c, []string{"name", "cloud-account-id", "realm-id"}, formatter)
+
+	floatingIPIn := map[string]interface{}{
+		"name":             c.String("name"),
+		"cloud_account_id": c.String("cloud-account-id"),
+		"realm_id":         c.String("realm-id"),
+	}
+
+	labelIDsByName, labelNamesByID := LabelLoadsMapping(c)
+
+	if c.IsSet("labels") {
+		labelsIdsArr := LabelResolution(c, c.String("labels"), &labelNamesByID, &labelIDsByName)
+		floatingIPIn["label_ids"] = labelsIdsArr
```

Comment:
In a strong type language it is an antipattern to include the type in the name of a variable. I'd rename this to labelIdsArr to labelIds or even go as far as reducing this to one line:
```Golang
		floatingIPIn["label_ids"] = LabelResolution(c, c.String("labels"), &labelNamesByID, &labelIDsByName)
```

Reply:
Done!